File: Client/src/app/models/player.ts
import { ClubPlayer } from "./clubPlayer";
import Nationality from "./nationality";
import Position from "./position";

export class Player {

    id: number = -1;
    photo: string = "";
    fullName: string = "";
    goals: number = -1;
    assists: number = -1;
    goalContributions: number = -1;
    titles: number = -1;
    appearances: number = -1;
    dateOfBirth: string = "";
    goalsRatio: number = -1;
    assistsRatio: number = -1;
    information: string = "";
    score: number = -1;
    positionId: number = -1;
    nationalityId: number = -1;
    position!:Position;
    nationality!:Nationality;
    clubPlayers: Array<ClubPlayer> = [];


}

File: Server/Server/Models/Inside/Player.cs

using Newtonsoft.Json.Serialization;

namespace Server.Models.Inside
{
    public class Player
    {

        [Key]
        public int Id { get; set; }
        public string Photo { get; set; } = string.Empty;
        public string FullName { get; set; } = string.Empty;
        public int Goals { get; set; }
        public int Assists { get; set; }
        public int GoalContributions { get; set; }
        public int Titles { get; set; }
        public int Appearances { get; set; }
        public string DateOfBirth { get; set; } = string.Empty;
        public double GoalsRatio { get; set; }
        public double AssistsRatio { get; set; }
        public string Information { get; set; } = string.Empty;
        public int Score { get; set; }
        public int PositionId { get; set; }
        public int NationalityId { get; set; }

        public virtual Position Position { get; set; } = null!;
        public virtual Nationality Nationality { get; set; } = null!;
        public virtual ICollection<ClubPlayer> ClubPlayers { get; set; } = null!;

        // image , ballondor,golden boot , chanpions league goals and assists,nantional goals and assists,word cup,
        // euro , copa america  
    }
}
